### DISCUSSION TITLE
How to enable smart merge control in Renovate Enterprise?

### DISCUSSION THREAD

**Discussion by mikaello** [mend/renovate-ce-ee#442] [Q&A] [answered]
We have Renovate Enterprise (running against a GitLab instance) and have enabled [merge confidence](https://docs.renovatebot.com/merge-confidence/), but would like to enable smart merge control. The guide [Configure your repository scan to use Smart Merge Control
](https://docs.mend.io/bundle/integrations/page/configure_your_repository_scan_to_use_smart_merge_control.html) says that the environment variable `MEND_ADVANCED_MERGE_CONFIDENCE_ENABLED=true` must be set, but not if it is in the server or worker. 

The guide also says that `renovate.json` is not supported for other platforms than GitHub:

> renovate.json file - (GitHub.com only) Standalone Renovate configuration - To point to an existing renovate.json file that includes your Smart Merge Control configuration, include the path to the renovate.json file within the `extends` block of your Renovate configuration located in your .whitesource or repo-config.json file.

I am a bit confused, also by all the information about the `.whitesource` configuration file which I have not seen before.

Is there an easy to follow guide for how to enable smart merge control for Renovate Enterprise users (also for users not using GitHub)?

**Comment by rarkins** [comment, marked as answer]
Hi @mikaello the docs you found are specific to a different Mend product called "repo integrations".

For Renovate use, you will need:
- A merge confidence token
- Use the matchConfidence option in packageRules: https://docs.renovatebot.com/configuration-options/#matchconfidence

I'll ensure you get the token as soon as possible! Most likely Jennifer will send to Andreas, is that ok? As a secondary priority we'll also update the Renovate EE docs so it's clear for anyone else in future.

Cc @justo-mend @nabeelsaabna

**Reply by mikaello** [reply]
Your suggestion is perfectly OK, and updated doc is also nice. Thanks!

**Reply by mikaello** [reply]
It would be nice with the documentation here. We also found something that seems a bit incorrect with the [`RENOVATE_X_MERGE_CONFIDENCE_API_BASE_URL` variable](https://docs.renovatebot.com/self-hosted-experimental/#renovate_x_merge_confidence_api_base_url). We have defined it as `https://artifactory.acme.com/artifactory/mend-developer-generic-remote`, but this errors:

```json
{
  "err": {
    "name": "HTTPError",
    "code": "ERR_NON_2XX_3XX_RESPONSE",
    "message": "Response code 401 (Unauthorized)",
    "stack": "HTTPError: Response code 401 (Unauthorized)\n    at Request.<anonymous> (/usr/src/app/node_modules/got/dist/source/as-promise/index.js:118:42)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)",
    "options": {
      "headers": {
        "user-agent": "RenovateBot/37.170.0 (https://github.com/renovatebot/renovate)",
        "authorization": "***********",
        "accept-encoding": "gzip, deflate, br"
      },
      "url": "https://artifactory.acme.com/artifactory/mend-developer-generic-remoteapi/mc/availability",
      "hostType": "merge-confidence",
      "username": "",
      "password": "",
      "method": "GET",
      "http2": false
    }
}
```

As you can see, the `api/mc/availability` part is added without slash. Is it by purpose that `RENOVATE_X_MERGE_CONFIDENCE_API_BASE_URL` must end with a trailing slash?

**Reply by nabeelsaabna** [reply]
it's part of the [renovate](https://github.com/renovatebot/renovate)
https://github.com/renovatebot/renovate/blob/7103ecd4023f163b0f51b1ffe05b667c126966a8/lib/util/merge-confidence/index.ts#L234-L257
will make sure to add an insure `/`

**Reply by Gabriel-Ladzaretti** [reply]
merged upstream
- https://github.com/renovatebot/renovate/pull/27536
